### PR TITLE
Forkchoice fix bug and cleanup

### PR DESCRIPTION
**Pull request by twoeths**
**Motivation**

- Clean up unnecessary functions/variables and flag
- Refactor forkchoice `nodeIsViableForHead()`

**Description**

Implements https://github.com/ethereum/consensus-specs/pull/3290

- Remove `bestJustifiedCheckpoint` functions and variables
- Remove `SAFE_SLOTS_TO_UPDATE_JUSTIFIED`
- Remove `computeUnrealizedFull` flag
- Refactor forkchoice `nodeIsViableForHead()`

### CODE REVIEW

File: packages/fork-choice/src/protoArray/protoArray.ts [hunk 1]
```diff
@@ -741,23 +733,66 @@ export class ProtoArray {
...
+    }
+
+    try {
+      const finalizedSlot = computeStartSlotAtEpoch(this.finalizedEpoch);
       const correctFinalized =
-        (nodeFinalizedEpoch === this.finalizedEpoch && nodeFinalizedRoot === this.finalizedRoot) ||
-        this.finalizedEpoch === 0;
+        this.finalizedRoot === this.getAncestor(node.blockRoot, finalizedSlot) || this.finalizedEpoch === 0;
       return correctJustified && correctFinalized;
+    } catch (_) {
```
Comment (dapplion) [inline packages/fork-choice/src/protoArray/protoArray.ts:760]: Can you refactor getAncestor to not throw to remove this blind try catch? Or scope the try catch to affect getAncestor call only

**Review by dapplion — COMMENTED** [review]
Overall looks good

File: packages/fork-choice/src/protoArray/protoArray.ts (hunk 1 above)
Comment (twoeths) [reply packages/fork-choice/src/protoArray/protoArray.ts:760]: `getAncestor()` is designed to throw so I don't want to break any existing flow. I added `getAncestorOrNull()` and use it here instead

**Comment by wemeetagain** [comment]
@tuyennhv @dapplion what's the status of this PR? don't want it to rot

**Review by dapplion — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.7.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.7.0) :tada:

**Pull request merged**